### DISCUSSION TITLE
Increasing width also cause the object to grow in size?

### DISCUSSION THREAD

**Discussion by Icyretsz** [fabricjs/fabric.js#10025] [Q&A]
Hi everyone,

I’m trying to implement a stroke width slider similar to the one in Canva. I use `Object.set('strokeWidth')` in the `onChange` method of the slider to modify the stroke width of an object. However, I've noticed that the stroke expands outwards, causing the overall size of the object to increase. 

Below is a video illustrating the issue.

https://github.com/user-attachments/assets/d6b053c2-e7ea-4a03-bf4b-6712384c0503

How can I adjust the stroke width while keeping the object size constant?

Thank you for your help!

**Comment by zhe-he** [comment]
Import the lib, and then just set the strokeWidth.

[demo](https://codesandbox.io/p/devbox/fabric-stroke-patch-nnyrlw?file=%2Fsrc%2Ftest-stroke-side.tsx%3A29%2C17)

**Comment by asturur** [comment]
@Icyretsz you can't.
What you can do is use center positioning ( set originX and originY as default to center ) to make it less strange.
Strokewidth expands half in and half out on the shape edge. that is how it is.

You could clip each shape with the exact same shape  to avoid seeing it, but that would still make the bounding box grow and would have bad performances probably
